mw/i2c_service/tests/test_adcsensor.cc

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file test_adcsensor.cc
+ * @author Michał Mańkowski (m.mankowski2004@gmail.com)
+ * @brief Unit tests for adcsensor
+ * @version 0.1
+ * @date 2024-06-14
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include "mw/i2c_service/controller/mock/mock_ads7828.h"
+#include "mw/i2c_service/controller/adcsensor/controller.hpp"
+
+class TestWrapper : public simba::i2c::ADCSensorController {
+ public:
+  float TestCalculateB(float R, float A, float A_MIN, float RES_MIN) {
+    return this->CalculateB(R, A, A_MIN, RES_MIN);
+  }
+  float TestCalculateA(float R, float RES_MAX, float RES_MIN, float A_MAX, float A_MIN) {
+    return this->CalculateA(R, RES_MAX, RES_MIN, A_MAX, A_MIN);
+  }
+  std::unordered_map<uint8_t, simba::i2c::SensorConfig> TestReadConfig(nlohmann::json data) {
+    return this->ReadConfig(data);
+  }
+  void TestSetConfig(std::unordered_map<uint8_t, simba::i2c::SensorConfig> db_) {
+    this->setConfig(db_);
+  }
+
+  void TestSetPtr(std::unique_ptr<simba::mock::MockADS7828> adc_) {
+    this->setPtr(std::move(adc_));
+  }
+};
+
+TEST(TestADCSensor, CalculateTest) {
```

Comment:
to mogło być sparametryzowane dla 2 różnych configów + chyba łatwiej by to było osobno sprawdzać A i B, piszesz działanie w python podajesz 3 zestawy danych i masz 3 scenariusze

Reply:
Pytałem się na pv, czy to parametryzować stwierdziłeś wtedy że nie

Reply:
ok